src/twisted/web/http.py

```diff
@@ -1586,7 +1593,8 @@ def loseConnection(self):
         """
         Pass the loseConnection through to the underlying channel.
         """
-        self.channel.loseConnection()
+        if self.channel:
```

Comment:
I guess this should be `if self.channel is not None:`, sorry if I missed that on an earlier review (if so feel free to leave this as-is if you prefer but maybe it's not a big deal since I asked for a news fragment change already. :smile: ).

Reply:
fixed in 4d7733c